backend/loader.py

```diff
@@ -30,6 +30,24 @@
 dir_path = os.path.dirname(__file__)
 
 
+def check_huggingface_component(component_name:str, cls_name:str, state_dict):
+    check_sd = False
+    comp_str = 'model'
+
+    if cls_name == 'AutoencoderKL':
+        check_sd = True
+        comp_str = 'VAE'
+    elif component_name.startswith('text_encoder') and cls_name in ['CLIPTextModel', 'CLIPTextModelWithProjection']:
+        check_sd = True
+        comp_str = 'CLIP'
+    elif cls_name == 'T5EncoderModel':
+        check_sd = True
+        comp_str = 'T5'
+
+    if check_sd and (not isinstance(state_dict, dict) or len(state_dict) <= 16):
```

Comment:
could you please explain this magic number 16? Why 16?

Reply:
@psydok to answer both questions, the `state_dict` variable and that specific condition, were both lifted from `load_huggingface_component()` which is called from `forge_loader()`.

These are Illyasviel code - I simply relocated this conditional check.  With this PR it is now checking the condition before trashing the currently loaded model data, so if an exception is raised the current model data does not get unloaded.

Reply:
Aside from the context clues of this condition, I otherwise have no idea what the expected structure of `state_dict` is supposed to be.  There is even less type hinting where I lifted the code from `load_huggingface_component()`

Reply:
Okay, I got it! Thank you very much!